actionpack/lib/action_dispatch/system_testing/browser.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActionDispatch
+  module SystemTesting
+    class Browser # :nodoc:
+      attr_reader :name
+      def initialize(name)
+        @name = name
+      end
+
+      def type
+        if name == :headless_chrome
+          :chrome
+        elsif name == :headless_firefox
+          :firefox
+        else
+          name
+        end
```

Comment:
This is picky, but I think this one and maybe the one below can be a case statement instead.